MVVM/ViewModel/RoverMode_UIOverlay.cs

```diff
@@ -19,6 +21,27 @@ public partial class RoverMode_UIOverlay : UIOverlay
 	public Task ControlModeChangedSubscriber(MqttClasses.ControlMode newMode)
 	{
 		ControlMode = (int)newMode;
+		CallDeferred(MethodName.UpdateSafeModeIndicatator);
 		return Task.CompletedTask;
 	}
+
+	public Task SettingsAppliedSubscriber()
+	{
+		CallDeferred(MethodName.UpdateSafeModeIndicatator);
+		return Task.CompletedTask;
+	}
+
+	void UpdateSafeModeIndicatator()
+	{
+		if (ControlMode == 1 && LocalSettings.Singleton.SpeedLimiter.Enabled == true && LocalSettings.Singleton.SpeedLimiter.MaxSpeed < 1)
+		{
+			SafeModeIndicator.Visible = true;
+			SafeModeIndicator.Text = $"Safe Mode ON - {Mathf.Round(LocalSettings.Singleton.SpeedLimiter.MaxSpeed * 100.0)}%";//Rounding may seem unnecessary, but without it numbers higher than 80 would be displayed as 79.99999999999999
+		}
+		else
+		{
+			SafeModeIndicator.Visible = false;
+		}
+	}
```

Comment:
This should look like this.

Notice different approach for floating point number conversion to string ;)

```suggestion
	public override void _Ready()
	{
		base._Ready();
		LocalSettings.Singleton.Connect(
			LocalSettings.SignalName.PropagatedPropertyChanged, 
			Callable.From<StringName, StringName, Variant, Variant>(OnSettingsPropertyChanged)
		);
	}

	void OnSettingsPropertyChanged(StringName category, StringName name, Variant oldValue, Variant newValue)
	{
		if (category != nameof(LocalSettings.SpeedLimiter)) 
			return;

		UpdateSafeModeIndicatator();
	}

	void UpdateSafeModeIndicatator()
	{
		if (
			ControlMode == (int)MqttClasses.ControlMode.Rover &&
			LocalSettings.Singleton.SpeedLimiter.Enabled &&
			LocalSettings.Singleton.SpeedLimiter.MaxSpeed < 1
		)
		{
			SafeModeIndicator.Visible = true;
			SafeModeIndicator.Text = $"Safe Mode ON - {(LocalSettings.Singleton.SpeedLimiter.MaxSpeed * 100.0).ToString("F0")}%";
		}
		else
		{
			SafeModeIndicator.Visible = false;
		}
	}
```

Reply:
Adressed in [3ebe073](https://github.com/wisniax/RoverControlApp/pull/16/commits/3ebe0731e1a8a9c9beb909a52183d7a8d1710188). Used .ToString("P") as it's a bit shorter and cleaner